src/channel/fs_prover_channel.rs

```diff
@@ -0,0 +1,112 @@
+use crate::channel::{Channel, ChannelStates, ProverChannel}; // ProverChannel 추가
+use crate::hashutil::TempHashContainer;
+use ark_ff::Field;
+use std::marker::PhantomData;
+
+use rand_chacha::rand_core::RngCore;
+use rand_chacha::ChaCha20Rng;
+
+pub struct FSProverChannel<F: Field, H: TempHashContainer> {
+    _ph: PhantomData<(F, H)>,
+    prng: ChaCha20Rng,
```

Comment:
We need to use the same PRNG as the stone-prover:

The transcript implementation(s) need to be 100% compatible with chose of the stone prover.

In this case the PRNGs used are covered in:

https://github.com/starkware-libs/stone-prover/blob/main/src/starkware/randomness/prng.cc

Reply:
So Blake2s, Keccak256, Poseidon (work-in-progreess)